### PR TITLE
Android nightly fixes

### PR DESCRIPTION
**Pull request by gunan**
This time tested for real:
http://ci.tensorflow.org/job/experimental-android/5/

PS: 25 minutes on a 32 CPU machine, wow this build is heavy!

### CODE REVIEW

**Review by andrewharp — APPROVED** [review]

File: tensorflow/tools/ci_build/ci_parameterized_build.sh [hunk 1]
```diff
@@ -501,14 +501,21 @@ fi
 
 chmod +x ${TMP_SCRIPT}
 
+# Map TF_BUILD container types to containers we actually have.
+if [[ "${CTYPE}" == "android_nightly" ]]; then
```
Comment (caisq) [inline tensorflow/tools/ci_build/ci_parameterized_build.sh]: LGTM in general. However, can we call it "android_nightly" something more informative and less tightly coupled with our particular CI setup? My understanding is that "android_nightly" is longer than "android" and runs (a lot of?) tests? So can we call it something like "android_test" or "android_full" and rename the old one ("andrdoid") to "android_build_only"?

Comment (andrewharp) [inline tensorflow/tools/ci_build/ci_parameterized_build.sh]: android_nightly doesn't run any explicit tests, but it does provide more complete build coverage and also exposes all the conceivable pre-built targets users may need.

Comment (caisq) [inline tensorflow/tools/ci_build/ci_parameterized_build.sh]: My nit point is the word "nightly" doesn't really tell us what the more extensive build does. It is also less flexible lest we decide to run the build in some cadence other than nightly (e.g., hourly or postsubmit).

**Comment by andrewharp** [comment]
s/android_nightly/android_full/ sgtm

I worry android_build_only is misleading because they both only build (for now), and in the future we may want to throw some quick tests into the presubmit. What about android_presubmit or android_quick?

**Comment by caisq** [comment]
+1 s/android_nightly/android_full/ 
For the quicker android build, I suggest we just keep the current name "android".

**Review by caisq — APPROVED** [review]
Can you run the experimental build again to make sure its working? Thanks.

**Comment by gunan** [comment]
http://ci.tensorflow.org/job/experimental-android/TF_BUILD_CONTAINER_TYPE=ANDROID_FULL,TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=NO_PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=cpu-slave/13/console

I am seeing some failures for my experimental build on this, but not sure what it means:

```
INFO: Reading 'startup' options from /var/lib/jenkins/workspace/experimental-android/TF_BUILD_CONTAINER_TYPE/ANDROID_FULL/TF_BUILD_IS_OPT/OPT/TF_BUILD_IS_PIP/NO_PIP/TF_BUILD_PYTHON_VERSION/PYTHON2/label/cpu-slave/bazel-ci_build-cache/.bazelrc: --batch
TF_BUILD_INFO = {container_type: "android", command: "/tmp/tf_build.sh", source_HEAD: "9bdea3de08242f40a0fd391e8c41726dd56cc478", source_remote_origin: "https://github.com/gunan/tensorflow.git", OS: "Linux", kernel: "4.4.0-36-generic", architecture: "x86_64", processor: "Intel(R) Xeon(R) CPU @ 2.50GHz", processor_count: "32", memory_total: "123776972 kB", swap_total: "0 kB", Bazel_version: "Build label: 0.4.2", Java_version: "1.8.0_111", Python_version: "2.7.6", gpp_version: "g++ (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4", swig_version: "", NVIDIA_driver_version: "", CUDA_device_count: "0", CUDA_device_names: "", CUDA_toolkit_version: ""}
/tmp/tf_build.sh: line 2: -c: command not found
```

@caisq how do we construct tf_build.sh? What should be there before `-c`?

**Pull request merged**